apps/src/music/analytics/AnalyticsReporter.ts

```diff
@@ -178,22 +184,30 @@ export default class AnalyticsReporter {
     );
   }
 
-  setProjectProperty(
-    property: (typeof trackedProjectProperties)[number],
-    value: string | number | undefined
+  setProjectProperty<K extends keyof ProjectContext>(
+    property: K,
+    value: ProjectContext[K]
```

Comment:
nice 😎 

Edit: oh but does this break anything for callers (e.g. in `useUpdateAnalytics`) since value could previously be `undefined`?

Reply:
My understanding is that this still works if its undefined (and I didn't see any TS compilation errors). eg, if the property is `levelType`, `ProjectContext[levelType]` returns `string | undefined`?